File: extract_pdf.js
const fs = require('fs');
const pdfModule = require('pdf-parse');

const pdfPath = 'C:\\Users\\alaco\\Downloads\\Коммерческое предложение.pdf';
const txtPath = 'c:\\Users\\alaco\\Academy_Webistan\\pdf_text.txt';

async function main() {
    try {
        let buffer = fs.readFileSync(pdfPath);
        let uint8Array = new Uint8Array(buffer.buffer, buffer.byteOffset, buffer.byteLength);
        const parser = new pdfModule.PDFParse(uint8Array);
        await parser.load();
        const result = await parser.getText();
        
        fs.writeFileSync(txtPath, result.text);
        console.log("PDF text extracted successfully to pdf_text.txt!");
    } catch (err) {
        console.error("Error:", err);
    }
}

main();

File: extract_pdf.py
import pypdf
import sys

def extract_pdf_text(pdf_path, txt_path):
    try:
        reader = pypdf.PdfReader(pdf_path)
        text = ""
        for i, page in enumerate(reader.pages):
            text += f"--- Page {i+1} ---\n"
            text += page.extract_text() + "\n"
        with open(txt_path, 'w', encoding='utf-8') as f:
            f.write(text)
        print(f"Successfully extracted text to {txt_path}")
    except Exception as e:
        print(f"Error: {e}")

if __name__ == "__main__":
    pdf_path = r"C:\Users\alaco\Downloads\Коммерческое предложение.pdf"
    txt_path = r"c:\Users\alaco\Academy_Webistan\pdf_text.txt"
    extract_pdf_text(pdf_path, txt_path)
